cql3/restrictions/statement_restrictions.cc

```diff
@@ -443,7 +443,7 @@ statement_restrictions::statement_restrictions(data_dictionary::database db,
     if (!expr::is_empty_restriction(_nonprimary_key_restrictions)) {
         if (_has_queriable_regular_index && _partition_range_is_simple) {
             _uses_secondary_indexing = true;
-        } else if (!allow_filtering) {
+        } else if (!allow_filtering && !type.is_delete() && !type.is_update()) {
```

Comment:
This condition looks suspicious to me: I think this case should be an error even for DELETE and UPDATE, just a different error message. Or do we already get an error in that case from some other code path? Maybe it's all correct, but I can't know for sure without a good test designed to check exactly this. I think a good test should:
1. Run `DELETE FROM tbl WHERE partitionKey = ? AND value = ?` and verify we get an error, but not the wrong one suggesting ALLOW FILTERING would work. Just like in Cassandra. If this request doesn't fail and instead just ignores the `value = ?` condition, it's a mistake.
2. Maybe Run `DELETE FROM tbl WHERE partitionKey = ? AND value = ? ALLOW FILTERING` and check it still doesn't work (it's a syntax error).
3. Maybe also run `DELETE FROM tbl WHERE partitionKey = ? clusteringKey=?` and see it still works, we didn't break it.

Reply:
Results:
1.`InvalidRequest: Error from server: code=2200 [Invalid query] message="Invalid where clause contains non PRIMARY KEY columns: value"`
2.`SyntaxException: line 1:55 : Syntax error`
3. I think you missed "AND" between partitionKey = ? and clusteringKey = ?. If that is the case - this command passes without errors.

Reply:
1. Good, I didn't realize that we already had the "Invalid where clause contains non PRIMARY KEY columns: value" check somewhere else.
2. Good.
3. Yes.
So good.